src/routes/safe/components/Settings/ManageOwners/RemoveOwnerModal/index.tsx

```diff
@@ -21,18 +22,14 @@ type OwnerValues = OwnerData & {
 export const sendRemoveOwner = async (
   values: OwnerValues,
   safeAddress: string,
-  safeVersion: string,
   ownerAddressToRemove: string,
   dispatch: Dispatch,
   txParameters: TxParameters,
+  connectedWalletAddress: string,
 ): Promise<void> => {
-  const gnosisSafe = getGnosisSafeInstanceAt(safeAddress, safeVersion)
-  const safeOwners = await gnosisSafe.methods.getOwners().call()
-  const index = safeOwners.findIndex(
-    (ownerAddress) => ownerAddress.toLowerCase() === ownerAddressToRemove.toLowerCase(),
-  )
-  const prevAddress = index === 0 ? SENTINEL_ADDRESS : safeOwners[index - 1]
-  const txData = gnosisSafe.methods.removeOwner(prevAddress, ownerAddressToRemove, values.threshold).encodeABI()
+  const sdk = await getSafeSDK(connectedWalletAddress, safeAddress)
+  const safeTx = await sdk.getRemoveOwnerTx(ownerAddressToRemove, +values.threshold)
+  const txData = safeTx.data.data
```

Comment:
Does this still work with `createTransaction()`?

Reply:
Yes, I have tested the 3 functionalities I've modified. Works like before.